### DISCUSSION TITLE
Changes in Optimization Settings may cause break in old input files (Plus New Feature)

### DISCUSSION THREAD

**Discussion by GabrielSoto-INL** [idaholab/HERON#298] [Releases and Patches]
Hello everyone!
I wanted to bring attention to a new feature for HERON input scripts that could potentially break previous scripts when using optimization settings.

## Changes to Optimization Settings

Previously, when running a HERON case in `opt` mode, a user could specify `<optimization_settings>` within the `<Case>` node. An optional entry within those settings was `<metric>` which allowed the user to select a statistic (e.g., "expectedValue", "valueAtRisk") applied to the set of NPV results from the inner runs. This is then used as the Outer objective function. 
``` xml
<!--old version-->
<Case name="Opt_Runs">
  <mode>opt</mode>
  <optimization_settings>
    <metric>expectedValue</metric>
  </optimization_settings>
  [...]
</Case>
```
For an old example of how this was used, [see here (an old version of the optimization_settings test)](https://github.com/idaholab/HERON/blob/afe1c79368e6fbbaed146e95c56017a9e2ff10ec/tests/integration_tests/mechanics/optimization_settings/heron_input.xml#L41).

We have recently switched to a format that differentiates between a statistical metric and economic metric to optimize. The old `<metric>` node is now called `<stats_metric>`. Additionally, we now allow for an `<opt_metric>` to be chosen besides NPV. HERON currently supports:
1. NPV
2. PI (Profitability Index)
3. IRR (Internal Rate of Return)

Levelized Cost (implemented as the `NPV_search` option in TEAL) is coming soon! The new format looks like this:
``` xml
<!--new version-->
<Case name="Opt_Runs">
  <mode>opt</mode>
  <optimization_settings>
    <stats_metric>expectedValue</stats_metric>
    <opt_metric>NPV</opt_metric>
  </optimization_settings>
  [...]
</Case>
```

## New Feature for Economic Metrics

There is also a new node in the `<Case>` nodes for returning additional economic metrics at the end of your run, regardless of `opt` or `sweep` mode. 
``` xml
<Case name="Opt_Runs">
  <economics>
    <EconMetrics>
      <NPV/>
      <IRR/>
      <PI/>
    </EconMetrics>
    [...]
  </economics>
  [...]
</Case>
```

These (along with all statistics in `<return_statistics>` applied) are reported back in the final CSV file at the end of the run.

**Comment by j-bryan** [comment]
Thanks for the heads up on this change, Gabe. I'll pass the word along to the guys at USU.
